server/lib/models/account.go

```diff
@@ -88,30 +95,30 @@ func calc_sha256(source string) string {
 }
 
 //compute_sign is calculate signature of login request.
-func compute_sign(id, secret string, walltime uint64) string {
-	src := strconv.FormatUint(walltime, 10) + id + secret
+func compute_sign(id uint64, secret string, walltime uint64) string {
+	src := strconv.FormatUint(walltime, 10) + strconv.FormatUint(id, 10) + secret
 	return calc_sha256(src)
 }
 
 //compute_user_secret calculates secret for users
-func compute_user_secret(id, pass string, walltime uint64) string {
-	return calc_sha256(strconv.FormatUint(walltime, 10) + id + pass + SECRET)
+func compute_user_secret(id uint64, pass string, walltime uint64) string {
+	return calc_sha256(strconv.FormatUint(walltime, 10) + strconv.FormatUint(id, 10) + pass + SECRET)
 }
 
 func (a *Account) ComputeSecret(pass string, walltime uint64) string {
-	return compute_user_secret(a.StringId(), pass, walltime)
+	return compute_user_secret(uint64(a.Id), pass, walltime)
 }
 
 func (a *Account) ComputeSign(walltime uint64) string {
-	return compute_sign(a.StringId(), a.Secret, walltime)
+	return compute_sign(uint64(a.Id), a.Secret, walltime)
 }
 
 func (a *Account) VerifySign(sign string, walltime uint64) bool {
 	computed := a.ComputeSign(walltime)
 	if sign == computed {
 		return true
 	}
-	log.Printf("invalid sign %v:%v:%v:%v:%v", sign, computed, a.StringId(), a.Secret, walltime)
+	log.Printf("invalid sign %v:%v:%v:%v:%v", sign, computed, a.Id, a.Secret, walltime)
```

Comment:
__[gometalinter]__
* [gotype] invalid operation: a (variable of type *Account) has no field or method Id
* [gotype] invalid operation: a (variable of type *Account) has no field or method Secret